File: elixir/022/022.exs
{:ok, contents} = File.read("../../files/names.txt")

names = contents |> String.split(",") |> Enum.map(fn name -> name |> String.replace("\"", "") end) |> Enum.sort

scores = names |> Enum.with_index |> Enum.map(fn {name, index} -> (name |> String.to_char_list! |> Enum.map(fn letter -> letter - 64 end) |> Enum.reduce(0, &(&1 + &2))) * (index + 1) end)

IO.puts scores
|> Enum.reduce(0, &(&1 + &2))
|> to_string

File: ruby/022/022.rb
contents = File.read(File.expand_path(File.join(__dir__, '..', '..', 'files', 'names.txt')))

names = contents.split(',').map { |name| name.gsub('"', '') }.sort

scores = names.map.with_index { |name, index| name.each_byte.map { |letter| letter - 64 }.inject(:+) * (index + 1) }

puts scores.inject(:+)
